cctp_primitives/src/commitment_tree/utils.rs

```diff
@@ -29,84 +34,95 @@ pub fn add_leaf(tree: &mut FieldElementsMT, leaf: &FieldElement, pos: &mut usize
     }
 }
 
-// Calculates hash of a sequentially concatenated data elements
-pub fn hash_vec(data: &Vec<FieldElement>) -> FieldElement {
-    let mut hasher = <FieldHash>::init(None);
-    for &fe in data {
-        hasher.update(fe);
+fn _get_root_from_field_vec(field_vec: Vec<FieldElement>, height: usize) -> Result<FieldElement, Error> {
+    assert!(height <= MHT_PARAMETERS.nodes.len());
+    if field_vec.len() > 0 {
+        let mut mt = new_mt(height)?;
+        for fe in field_vec.into_iter(){
+            mt.append(fe);
+        }
+        mt.finalize_in_place();
+        mt.root().ok_or(Error::from("Failed to compute Merkle Tree root"))
+
+    } else {
+        Ok(MHT_PARAMETERS.nodes[height])
     }
-    hasher.finalize()
 }
 
-// Generates vector of random bytes
-pub fn rand_vec(len: usize) -> Vec<u8> {
-    let mut rng = rand::thread_rng();
-    (0.. len).map(|_|rng.gen()).collect()
+/// Get the Merkle Root of a Binary Merkle Tree of height 12 built from the Backward Transfer list
+pub fn get_bt_merkle_root(bt_list: &[(u64,[u8; 20])]) -> Result<FieldElement, Error>
```

Comment:
Modify implementation according to this: https://github.com/HorizenOfficial/zendoo-sc-cryptolib/blob/development_tmp/api/src/ginger_calls.rs#L177